data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttesterSlashings.java

```diff
@@ -13,36 +13,55 @@
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition.listOf;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.core.util.Header;
 import io.javalin.http.Context;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
 import java.util.List;
+import java.util.function.Function;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.NodeDataProvider;
 import tech.pegasys.teku.api.response.v1.beacon.GetAttesterSlashingsResponse;
-import tech.pegasys.teku.api.schema.AttesterSlashing;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.IndexedAttestation;
 
-public class GetAttesterSlashings extends AbstractHandler {
+public class GetAttesterSlashings extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/beacon/pool/attester_slashings";
   private final NodeDataProvider nodeDataProvider;
 
-  public GetAttesterSlashings(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
-    this.nodeDataProvider = dataProvider.getNodeDataProvider();
+  public GetAttesterSlashings(final DataProvider dataProvider, Spec spec) {
+    this(dataProvider.getNodeDataProvider(), spec);
   }
 
-  GetAttesterSlashings(final NodeDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  GetAttesterSlashings(final NodeDataProvider provider, Spec spec) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getAttesterSlashings")
+            .summary("Get Attester Slashings")
+            .description(
+                "Retrieves attester slashings known by the node but not necessarily incorporated into any block.")
+            .tags(TAG_BEACON)
+            .response(
+                SC_OK,
+                "Request successful",
+                getResponseType(spec.getGenesisSpecConfig())) // TODO check correct
```

Comment:
^^ Just need to check this is the correct approach to getting the `SpecConfig`

Reply:
I'd probably pass just the spec and then if we need to get different configs at a future date it'll just be an implementation detail, but this is fine too...

Reply:
either way the TODO needs to disappear :)